abis_mapping/types/schema.py

```diff
@@ -7,31 +7,95 @@
+    """Field model of a schema.
+
+    The properties of a field consisting of those properties used by frictionless for performing validations as well
+    as extras to assist with the looking up of vocabularies when mapping as well as assisting with the creation
+    of instruction documentation.
+    [Frictionless reference](https://specs.frictionlessdata.io/table-schema).
+    """
+
+    name: Annotated[
+        str,
+        pydantic.Field(
+            description="[Required by frictionless](https://specs.frictionlessdata.io/table-schema/#name). The field descriptor MUST contain a name property. This property SHOULD correspond to the name of field/column in the data file (if it has a name). As such it SHOULD be unique (though it is possible, but very bad practice, for the data file to have multiple columns with the same name). name SHOULD NOT be considered case sensitive in determining uniqueness. However, since it should correspond to the name of the field in the data file it may be important to preserve case.",
+        ),
+    ]
+    title: Annotated[
+        str,
+        pydantic.Field(
+            description="[Frictionless reference](https://specs.frictionlessdata.io/table-schema/#title). A human readable label or title for the field.",
+        ),
+    ]
+    description: Annotated[
+        str,
+        pydantic.Field(
+            description='[Frictionless reference](https://specs.frictionlessdata.io/table-schema/#description). A description for this field e.g. "The recipient of the funds".',
+        ),
+    ]
+    example: Annotated[
+        str | None,
+        pydantic.Field(
+            description="[Frictionless reference](https://specs.frictionlessdata.io/table-schema/#example). An example value of the field",
+        ),
+    ] = None
+    type: Annotated[
+        str,
+        pydantic.Field(
+            description="[Frictionless reference](https://specs.frictionlessdata.io/table-schema/#types-and-formats). `type` and `format` properties are used to give the type of the field. A fields `type` property is a string indicating the type of this field.",
+        ),
+    ]
+    format: Annotated[
+        str | None,
+        pydantic.Field(
+            description="[Frictionless reference](https://specs.frictionlessdata.io/table-schema/#types-and-formats). `type` and `format` properties are used to give the type of the field. A field's `format` property is a string, indicating a format for the field type.",
+        ),
+    ]
+    url: Annotated[pydantic.AnyUrl | None, pydantic.Field(description="The IRI of the field's concept.")] = None
     constraints: Constraints
-    vocabularies: list[str] = []
+    vocabularies: Annotated[
+        list[str],
+        pydantic.Field(
+            description="Optional list of vocabulary IDs, defined internally within the project. Provided IDs need to have been registered to be valid. See [`abis_mapping.vocabs`](/abis_mapping/vocabs/)."
+        ),
+    ] = []
```

Comment:
improvement - swap `= []` for `default_factory=list` to avoid having a shared mutable default

Reply:
implemented